### ISSUE TITLE
Add testing of external components on more platforms

### ISSUE THREAD

**Issue by baentsch** [openssl/openssl#20679]
[Testing of external components](https://github.com/openssl/openssl/blob/master/test/README-external.md) has only really been done for [Linux variants](https://github.com/open-quantum-safe/oqs-provider/issues/132#issuecomment-1494568386). This issue is to suggest making them pass and adding them to CI for more platforms, e.g., moving a discussion from [here](https://github.com/open-quantum-safe/oqs-provider/issues/132) to this issue / a future PR.

**Comment by t8m** [comment]
This would be certainly nice to have. On the other hand the testing of external components with Linux provides most of the benefits and although testing them on additional platforms is surely useful, I do not expect that many problems to be found by this. So not a critical priority but we would certainly take patches that make the external testing on additional platforms work (+ corresponding on-push CI runs).

**Comment by mouse07410** [comment]
> the testing of external components with Linux provides most of the benefits

Sorry, I don't think I understand it. Unless spoken from a Linux vendor point of view (which I didn't think you were)?

> and although testing them on additional platforms is surely useful, I do not expect that many problems to be found by this

Respectfully disagree - unless you intend to have Linux as the only "fully-supported" platform, and consider other platforms as "supported if/when it works" - like with apps written for Linux that may or may not compile and run elsewhere.

**Comment by t8m** [comment]
I do not think this is going to be added. It is already quite demanding to maintain the existing external tests on Linux.
